src/test/java/com/example/project/test/ShoppingListTest.java

```diff
@@ -0,0 +1,138 @@
+package com.example.project.test;
+
+import com.example.project.page_object.AllListsPage;
+import com.example.project.page_object.ListPage;
+import io.appium.java_client.android.AndroidDriver;
+import io.appium.java_client.android.AndroidElement;
+import org.openqa.selenium.By;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+public class ShoppingListTest {
+
+    AndroidDriver<AndroidElement> driver;
+
+    @BeforeClass
+    public void setUp() throws MalformedURLException {
+        DesiredCapabilities cap = new DesiredCapabilities();
+        cap.setCapability("deviceName", "sdk_gphone_x86_arm");
+        cap.setCapability("udid", "emulator-5554");
+        cap.setCapability("platformName", "Android");
+        cap.setCapability("platformVersion", "11");
+
+        cap.setCapability("appPackage", "com.slava.buylist");
+        cap.setCapability("appActivity", "com.slava.buylist.MainActivity");
+
+        URL url = new URL("http://127.0.0.1:4723/wd/hub");
+        driver = new AndroidDriver<>(url, cap);
+
+        driver.manage().timeouts().implicitlyWait(15, TimeUnit.SECONDS);
+    }
+
+    @BeforeMethod
+    public void resetApp() {
+        driver.resetApp();
+        driver.findElementById("com.android.permissioncontroller:id/continue_button").click();
+        driver.findElementById("android:id/button1").click();
+    }
+
+    @Test
+    public void createNewList_successful() throws InterruptedException {
+        final String listName = "New list";
+        AllListsPage allListsPage = createNewList(listName);
+        Thread.sleep(2000);
+        driver.hideKeyboard();
+        driver.navigate().back();
+        assertThat(allListsPage.firstListNameTextView.getText()).isEqualTo(listName);
+    }
+
+    @Test
+    public void addItemToList_successful() {
+        createNewList("New list");
+
+        waitForListPageToLoad();
+
+        ListPage listPage = addItemToList("Milk", "1.2", "2", "4", "2");
```

Comment:
Let's imagine that one needs to set more item fields then she will need to rework all those methods. So it would be more handy to create a dedicated `Item` class and operate with it.

Reply:
Good point! :)